### PR TITLE
Add GlobeAnchor

### PR DESCRIPTION
**Pull request by kring**
This is a cesium-native implementation of type of functionality that is found in the `CesiumGlobeAnchorComponent` in Cesium for Unreal and the `CesiumGlobeAnchor` `MonoBehaviour` in Cesium for Unity. By providing the core functionality in cesium-native, we avoid reimplementing this somewhat tricky code in every game engine.

CesiumGS/cesium-unity#217 makes Cesium for Unity use this class. Switching Cesium for Unreal to use it will be a future feature, perhaps as part of CesiumGS/cesium-unreal#729.

There are also some smaller changes in this PR:

* Renamed `CesiumGeometry::AxisTransforms` to simply `Transforms`.
* Renamed `CesiumGeospatial::Transforms` to `GlobeTransforms`.
* Added `createTranslationRotationScaleMatrix` and `computeTranslationRotationScaleFromMatrix` to `CesiumGeometry::Transforms`

### CODE REVIEW

File: CesiumGeometry/test/TestAxisTransforms.cpp [hunk 1]
```diff
@@ -1,45 +1,45 @@
-#include "CesiumGeometry/AxisTransforms.h"
+#include "CesiumGeometry/Transforms.h"
```
Comment (j9liu) [inline CesiumGeometry/test/TestAxisTransforms.cpp:1]: This should probably be renamed to `TestTransforms.cpp` for consistency. I'll push a fix.

File: CesiumGeospatial/include/CesiumGeospatial/GlobeAnchor.h [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+#include <glm/mat4x4.hpp>
+
+#include <optional>
+
+namespace CesiumGeospatial {
+
+class LocalHorizontalCoordinateSystem;
+
+/**
+ * @brief Anchors an object to the globe by definining a transformation from the
```
Comment (j9liu) [inline CesiumGeospatial/include/CesiumGeospatial/GlobeAnchor.h:16]:
```suggestion
 * @brief Anchors an object to the globe by defining a transformation from the
```

I'll push a fix for this typo.

**Review by j9liu — COMMENTED** [review]
@kring looks good, I just had one question to make sure of my understanding

File: CesiumGeometry/src/Transforms.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    double lengthColumn2 = glm::length(rotationAndScale[2]);
+
+    glm::dmat3 rotationMatrix(
+        rotationAndScale[0] / lengthColumn0,
+        rotationAndScale[1] / lengthColumn1,
+        rotationAndScale[2] / lengthColumn2);
+
+    glm::dvec3 scale(lengthColumn0, lengthColumn1, lengthColumn2);
+
+    glm::dvec3 cross = glm::cross(rotationAndScale[0], rotationAndScale[1]);
```
[the remark this answers is not in the document]
Comment (kring) [reply CesiumGeometry/src/Transforms.cpp:102]: I believe it is necessary to support negative scales, which are in turn necessary to transform the left-handed coordinate systems used by Unity and Unreal.

[the remark this answers is not in the document]
Comment (j9liu) [reply CesiumGeometry/src/Transforms.cpp:102]: I see, that makes sense!

**Comment by j9liu** [comment]
thanks @kring!

**Comment by j9liu** [comment]
ah, I forgot to say that the new globe anchor class should probably be mentioned in `CHANGES.md`! but we can just push a commit to main

**Comment by kring** [comment]
> ah, I forgot to say that the new globe anchor class should probably be mentioned in CHANGES.md! but we can just push a commit to main

Oops, not sure I managed to add all that other stuff to CHANGES.md, but forgot to mention that main point of the PR! I've added it in main.

**Pull request merged**